crates/diesel_models/src/authorization.rs

```diff
@@ -57,6 +57,20 @@ pub struct AuthorizationUpdateInternal {
     pub connector_authorization_id: Option<String>,
 }
 
+impl AuthorizationUpdateInternal {
+    pub fn create_authorization(self, source: Authorization) -> Authorization {
+        Authorization {
+            status: self.status.unwrap_or(source.status),
+            error_code: self.error_code,
+            error_message: self.error_message,
+            modified_at: self.modified_at.unwrap_or(source.modified_at),
```

Comment:
Can we please get the current time instead of using source.modified_at (common_utils::date_time::now())